File: front/src/models/checkIn.ts
import { Like } from './like';

export interface CheckIn {
  id: number;
  beerId: number;
  rating: number;
  notes: string;
  date: string;
  userId: string;
}

export interface CommentDTO {
  id: number;
  text: string;
  createdAt: Date;
  checkInId: number;
  userId: string;
  username: string;
}

export interface CheckInDTO {
  id: number;
  beerId: number;
  beerName: string;
  rating: number;
  notes?: string;
  date: string;
  userId: string;
  firstName: string;
  lastName: string;
  beerImageUrl: string;
  brewery: string;
  country: string;
  likes: Like[];
  isLikedByCurrentUser: boolean;
  comments: CommentDTO[];
}

File: backend/Models/CheckIn.cs
namespace backend.Models
{
    public class CheckIn
    {
        public int Id { get; set; } 
        public int BeerId { get; set; }
        public virtual Beer Beer { get; set; }
        public required decimal Rating { get; set; }
        public string Brewery { get; set; }
        public string? Notes { get; set; } 
        public DateTime Date { get; set; } = DateTime.UtcNow;
        public string UserId { get; set; }
        public virtual User User { get; set; }
        public ICollection<Like> Likes { get; set; }
        public ICollection<Comment> Comments {get; set;}
    }
}